### PR TITLE
Release v3.5.0

### PR DESCRIPTION
**Pull request by pabera**
## New features
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2179
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2177
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2204
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2169
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2190
* [x] Improved `armv6` support
  * [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2217
* [x] Github CI Workflow improvements
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2233
* [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2218
* Documentation Improvements
  * [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2216
  * [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2225
* Other Bugfixes  
  * [x] https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2235

### CODE REVIEW

File: .github/workflows/pythonpackage_future3.yml [hunk 1]
```diff
@@ -39,9 +39,9 @@ jobs:
         pip3 install -r src/jukebox/components/rfid/hardware/pn532_i2c_py532/requirements.txt
         pip3 install -r src/jukebox/components/rfid/hardware/rdm6300_serial/requirements.txt
         pip3 install -r src/jukebox/components/rfid/hardware/rc522_spi/requirements.txt
-    - name: Run pytest
+    - name: Run pytest with coverage
       run: |
-        ./run_pytest.sh --cov --cov-report xml
+        ./run_pytest.sh --cov --cov-report xml --cov-config=.coveragerc
```
Comment (Groovylein) [inline .github/workflows/pythonpackage_future3.yml:44]: What do you expect by running it with coverage?

**Comment by AlvinSchiller** [comment]
Whats missing:
The need to update the jukebox.yaml for the coverart should be documented [here](https://github.com/MiczFlor/RPi-Jukebox-RFID/blob/future3/develop/documentation/builders/update.md).
Or we implement the automatic update / default value initialization before (which i would prefer :) ).

File: documentation/builders/gpio.md [hunk 2]
```diff
@@ -0,0 +1,430 @@
+# GPIO Recipes
+
+## Enabling GPIO
+
+The GPIO module needs to be enabled in your main configuration file ``shared/settings/jukebox.yaml``. Look for the
```
Comment (Groovylein) [inline documentation/builders/gpio.md:5]: The last 'the' might be too much 😉

File: documentation/developers/development-environment.md [hunk 3]
```diff
@@ -55,3 +56,14 @@ pip install pyzmq
 ```
 
 You will have to start Jukebox core application and the WebUI separately. The MPD usually runs as a service.
+
+### Using WSL
+
+You can also use WSL on Windows 10 or 11. This section describes how to use WSL with Visual Studio Code.
+
+1. Install a Debian or Ubuntu image from Microsoft Store
```
Comment (Groovylein) [inline documentation/developers/development-environment.md:64]: I think you need to install wsl as well. I would provide a link here:
https://learn.microsoft.com/en-us/windows/wsl/install

File: src/jukebox/jukebox/version.py [hunk 4]
```diff
@@ -1,14 +1,16 @@
 
 VERSION_MAJOR = 3
-VERSION_MINOR = 4
+VERSION_MINOR = 5
 VERSION_PATCH = 0
-VERSION_EXTRA = ""
+VERSION_EXTRA = "alpha"
```
Comment (Groovylein) [inline src/jukebox/jukebox/version.py:5]: Still alpha? Should we change it to beta?

**Comment by s-martin** [comment]
Event device is not merged yet

File: src/jukebox/jukebox/version.py (hunk 4 above)
Comment (AlvinSchiller) [reply src/jukebox/jukebox/version.py:5]: We just use "alpha" to differentiate the develop branch. Theres no "beta", after the merge its prod on main.

**Comment by pabera** [comment]
> Event device is not merged yet

I know - it's considered a draft and just created to represent the next release early on (updated the description)

**Review by s-martin — APPROVED** [review]

File: .github/workflows/pythonpackage_future3.yml (hunk 1 above)
Comment (s-martin) [reply .github/workflows/pythonpackage_future3.yml:44]: In coveralls.io you can check how much of the code is covered by our unit tests.
The Python GitHub Action runs the tests with coverage.

See the coveralls comment in this PR or other out related PRs.

Comment (Groovylein) [reply .github/workflows/pythonpackage_future3.yml:44]: OK, do we look for more coverage? Should we update the Contribution info as well?

Comment (s-martin) [reply .github/workflows/pythonpackage_future3.yml:44]: I think generally it's useful to improve the coverage, but from my point of view we should focus on making future3 feature "complete" at this point.

But it probably wouldn't hurt to encourage people in CONTRIBUTING to add unit tests.

**Comment by AlvinSchiller** [comment]
I think we have a problem on the update path with this release.
If currently described "just pull" isn't enough. And a rebuild of the webapp is also not possible if node is not installed in main.

Maybe we find a quickfix, so a rerun of the installation is possible, without massive changes?
Ideas: 
* Move to current Project Folder to "*-backup-$datetime" 
* Let the user rerun the installation
* copy 'shared'-folder from backup to the current installation
* give note that manually changes need to be transfered manually

A proper way can then be implemented with v3.6.0

**Comment by AlvinSchiller** [comment]
> I think we have a problem on the update path with this release. The currently described "just pull" method isn't enough. And a rebuild of the webapp is also not possible if node is not installed in main.
> 
> Maybe we find a quickfix, so a rerun of the installation is possible, without massive changes? Ideas:
> 
>     * Move to current Project Folder to "*-backup-$datetime"
> 
>     * Let the user rerun the installation
> 
>     * copy 'shared'-folder from backup to the current installation
> 
>     * give note that manually changes need to be transfered manually
> 
> 
> A proper way can then be implemented with v3.6.0

Or we just state that updates are currently broken and a fresh installation has to be done.

**Review by AlvinSchiller — APPROVED** [review]

**Comment by AlvinSchiller** [comment]
Test errors for Web App download can be ignored. This is caused due to the version bump, as no bundle is build for a release version on develop.
Well be fixed after the release and the bump to the next develop version.

**Pull request merged**